index.js

```diff
@@ -1,40 +1,100 @@
 const fs = require('fs-extra');
 
-
 const getAllFilteredItems = (path, isAllDirectory, isAllFiles) => {
 
-    return (fs.readdirSync(path) || []).filter(item => {
-        const isDirectory = fs.statSync(`${path}/${item}`).isDirectory();
-        return (isDirectory && isAllDirectory || !isDirectory && isAllFiles);
-    });
+  return (fs.readdirSync(path) || []).filter(item => {
+    const isDirectory = fs.statSync(`${path}/${item}`).isDirectory();
+    return (isDirectory && isAllDirectory || !isDirectory && isAllFiles);
+  });
 };
 
-const copyItems = ({items, destination, sourceDirPath}) =>
-    items.forEach(file =>
-        fs.copySync(`${sourceDirPath}/${file}`, `${destination}/${file}`)
-    );
+const copyItemsSync = ({
+    items,
+    destination,
+    sourceDirPath
+  }) =>
+  items.forEach(item =>
+    fs.copySync(`${sourceDirPath}/${item}`, `${destination}/${item}`)
+  );
 
 const copyFilesSync = filesData => {
 
-    try {
-        Object.keys(filesData).forEach(sourceDirPath => {
+  try {
+    Object.keys(filesData).forEach(sourceDirPath => {
 
-                const folderObject = filesData[sourceDirPath];
-                const {options: {allFiles, allDirectories} = {}, destination, foldersAndFiles = []} = folderObject;
+      const folderObject = filesData[sourceDirPath];
+      const {
+        options: {
+          allFiles,
+          allDirectories
+        } = {},
+        destination,
+        foldersAndFiles = []
+      } = folderObject;
 
-                const items = [...foldersAndFiles, ...getAllFilteredItems(sourceDirPath, allDirectories, allFiles)];
+      const items = [...foldersAndFiles, ...getAllFilteredItems(sourceDirPath, allDirectories, allFiles)];
 
-                copyItems({items, destination, sourceDirPath});
-            }
-        );
-    } catch (error) {
-        console.error(error);
-        return error;
-    }
+      copyItemsSync({
+        items,
+        destination,
+        sourceDirPath
+      });
+    });
+  } catch (error) {
+    console.error(error);
+    return error;
+  }
 };
 
-module.exports = {
-    copyFilesSync,
-};
+const copyItemsAsync = ({
+    items,
+    destination,
+    sourceDirPath
+  }) =>
+  items.map(item =>
+    new Promise((resolve, reject) => {
+      fs.copy(`${sourceDirPath}/${item}`, `${destination}/${item}`, err => {
+        if (err)
+          reject(err);
+        resolve();
+      });
+    })
+  );
+
+
+const copyFiles = async filesData => {
 
+  try {
+    const promisesArrays = Object.keys(filesData).map(sourceDirPath => {
```

Comment:
can be better if you will change it to `for` loop.
this way you could save iteration.

Reply:
please do it in `copyFilesSync` too